src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -84,15 +86,50 @@ class UserService(val directoryDAO: DirectoryDAO, val googleDirectoryDAO: Google
     } yield deleteResult
   }
 
+  def createUserPetServiceAccount(user: WorkbenchUser): Future[WorkbenchUserPetServiceAccountEmail] = {
+    def create() = {
+      val (petSa, petSaDisplayName) = toPetSAFromUser(user)
+      for {
+        petServiceAccount <- googleIamDAO.createServiceAccount(petServiceAccountConfig.googleProject, petSa, petSaDisplayName)
+        _ <- googleDirectoryDAO.addMemberToGroup(WorkbenchGroupEmail(toProxyFromUser(user.id.value)), petServiceAccount.email)
+        _ <- Future.traverse(petServiceAccountConfig.serviceAccountActors) { email =>
```

Comment:
What happens when we later need to add (or remove) a service account actor from the Sam conf list? Do we do the migration manually? Or some kinda boot-time fixup?

Reply:
Good question. I guess one thing we could do is when this endpoint is called and an existing pet SA is found, recheck the list of actors and add any that are missing. Kind of a lazy migration. Removals are trickier but that could be a manual migration.

Reply:
can we use a google group that we can maintain manually? If not, I would just leave it as a manual migration given that this will be infrequent.

Reply:
I'll punt this for now, and assume manual intervention will be needed. AFAIK the only pet service account actor we'll have initially is the Leo service account.

Reply:
OK, but expect Cromwell to likely come along soon(ish).